### PR TITLE
effect add versioning

### PR DESCRIPTION
**Pull request by star-e**
Extension pr: https://github.com/cocos/engine-extensions/pull/418
Engine supports compressed/uncompressed effect.bin

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

## Greptile Summary

The pull request introduces versioning for effects and includes changes to the `layout-graph-editor.ts` file to add a `getLayoutGraphDataVersion` function.

- Added `getLayoutGraphDataVersion` function in `cocos/rendering/custom/layout-graph-editor.ts` to return a version number for layout graph data.
- No other significant changes or issues identified in the file.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
   },
   "include": [
       "exports/**/*.ts",
+      "editor/exports/**/*.ts",
```
Comment (star-e) [inline tsconfig.json:56]: Fix github ESLint error.

File: cocos/rendering/custom/index.jsb.ts [hunk 2]
```diff
@@ -39,6 +39,8 @@ let _pipeline: Pipeline | null = null;
 export const INVALID_ID = 0xFFFFFFFF;
 export const enableEffectImport = true;
 
+const LAYOUT_HEADER_SIZE = 8;
```
Comment (minggo) [inline cocos/rendering/custom/index.jsb.ts:42]: It is 4 for 8? As `INVALID_ID` is 0xFFFFFFFF, seems it is 4 bytes.

File: cocos/rendering/custom/index.jsb.ts [hunk 3]
```diff
@@ -62,10 +64,17 @@ export function getCustomPipeline (name: string): PipelineBuilder {
 }
 
 export function init (device: Device, arrayBuffer: ArrayBuffer | null) {
-    if (arrayBuffer) {
-        const inflator = new zlib.Inflate(new Uint8Array(arrayBuffer));
-        const decompressed = inflator.decompress() as Uint16Array;
-        _renderModule = render.Factory.init(device, decompressed.buffer);
+    if (arrayBuffer && arrayBuffer.byteLength >= LAYOUT_HEADER_SIZE) {
+        const header = new DataView(arrayBuffer, 0, LAYOUT_HEADER_SIZE);
+        if (header.getUint32(0) === INVALID_ID) {
```
Comment (minggo) [inline cocos/rendering/custom/index.jsb.ts:69]: LAYOUT_HEADER_SIZE is 8,  but uses header.getUint32() to get 4 bytes to compare?

**Review by dumganhar — APPROVED** [review]

File: cocos/rendering/custom/index.jsb.ts (hunk 2 above)
Comment (star-e) [reply cocos/rendering/custom/index.jsb.ts:42]: 4 byte 0xFFFFFFFF + 4 byte version. It's totally 8 bytes.

Comment (minggo) [reply cocos/rendering/custom/index.jsb.ts:42]: Got it.

File: cocos/rendering/custom/index.jsb.ts (hunk 3 above)
Comment (star-e) [reply cocos/rendering/custom/index.jsb.ts:69]: Only first 4 bytes are used to check compressed/uncompressed. The other 4 bytes are used as version, which is not used in engine. In editor, version is used to upgrade effect.bin.

**Review by minggo — APPROVED** [review]

**Pull request merged**